docs/architecture/ADR-009-liquidity-mining

```diff
@@ -0,0 +1,176 @@
+# ADR 009: Liquidity Mining
+
+## Changelog
+
+- May 1 2022: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+Umee wishes to add support for liquidity mining incentives; i.e. additional rewards on top of the normal `x/leverage` lending APY for supplying base assets.
+
+The following is a record of preferences and/or requirements collected during initial exploration:
+
+| Q | A |
+| - | - |
+| Where do the reward tokens come from? | The community pool (which can be funded) but at a fixed initial amount. |
+| Are rewards liquid? | Yes. |
+| How do users participate? | Lockup uTokens from lending. |
+| What are the lockup periods? | Tiers: 24h/14d/30d. Three tiers typically provide a clean UX. Periods would be difficult to change post-launch |
```

Comment:
Lockup periods were required when `Epoch` implementation was going to be used (to prevent lending right at the time of the epoch and withdrawing right after).

Now that epoch is gone, we have two reasonable choices:
- Keep lock durations, which may help defend against "run on the bank" scenarios
- Abandon lock/unlock durations and tiers, and directly incentivize all lending

That discussion can go in the Alternatives section once we figure out what to do

Reply:
No, I mean I found a computationally efficient way to get the reward structure we specced out without using an epoch and iterating over accounts during EndBlocker. Acts a lot a pared down version of how staking reward math works, but requires a "claim" action.

I still need to add that section to the ADR, but once it's in we can figure out if it's good.

Reply:
Cool. I was developing a similar algorithm in Go in the past, probably could be adapted. It was for rewards streaming. So we can probably ship it pretty fast.